include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -0,0 +1,135 @@
+        }
+
+        /**
+         * @brief Makes prepared statement from function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) volatile) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        /**
+         * @brief Makes prepared statement from function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) const) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        /**
+         * @brief Makes prepared statement from function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        /**
+         * @brief Makes prepared statement from lambda's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename CallableClass>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, CallableClass&&) {
+            // This is actually only callable argument type deductor. We don't care about callable type, so we can decay Callable object
+            return generatePreparedStatementImpl(connection, query, &std::decay_t<CallableClass>::operator());
+        }
+
+        /**
+         * @brief Makes prepared statement from plain old C function's signature
+         * @param connection Connection handle into database
+         * @param query Query for prepared statement
+         */
+        template <typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+    }
+
+    /**
+     * @brief Executes query, reads input data and passes them to callback function
```

Comment:
Describe please how it is passed - i mean that PS's result columns is passed to provided processingFunction - for every result row prcessingFunction is called once and number of result columns and processingFunction arguments types and order must correspond to themselves.